services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/tormaystarkastelu/TormaystarkasteluController.kt

```diff
@@ -0,0 +1,106 @@
+
+import fi.hel.haitaton.hanke.HankeError
+import fi.hel.haitaton.hanke.domain.Hanke
+import fi.hel.haitaton.hanke.validation.ValidHanke
+import mu.KotlinLogging
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.*
+
+
+private val logger = KotlinLogging.logger { }
+
+@RestController
+@RequestMapping("/tormays")
+@Validated
+class TormaystarkasteluController {
+
+    @GetMapping("/{hankeTunnus}")
+    fun getTormaysTarkastelu(@PathVariable(name = "hankeTunnus") hankeTunnus: String?): ResponseEntity<Any> {
+        logger.info {
+            "Creating tormaystarkastelu for hanke: ${hankeTunnus}"
+        }
+
+        if (hankeTunnus == null) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(HankeError.HAI1002) //TODO: own error code?
+        }
+
+        return try {
+            //call service
+
+            val tormaysResults = getDummyTormaystarkasteluTulos()
+
+
+            // has saved tormaystulos to database
+            //return hanke with tormaystulos
+            logger.info { "tormaystarkastelu created for Hanke ${hankeTunnus}." }
+            ResponseEntity.status(HttpStatus.OK).body(tormaysResults)
+        } catch (e: Exception) {
+            logger.error(e) {
+
+                HankeError.HAI1006.toString()
+
+            }
+            ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(HankeError.HAI1006)
+        }
+    }
+
+    private fun getDummyTormaystarkasteluTulos(): TormaystarkasteluTulos {
+        val tormaysResults = TormaystarkasteluTulos()
+        tormaysResults.hankeId = 2 //TODO: selvitettävä jostain
+        tormaysResults.joukkoliikenneIndeksi = 3.4
+        tormaysResults.pyorailyIndeksi = 4.2
+        tormaysResults.perusIndeksi = 2.1
+
+        tormaysResults.liikennehaittaIndeksi = LiikennehaittaIndeksiType()
+        tormaysResults.liikennehaittaIndeksi!!.indeksi = 4.2
+        tormaysResults.liikennehaittaIndeksi!!.type = IndeksiType.PYORAILYINDEKSI
+
+        //TODO: in the future this would contain "Viereiset hankkeet" too
+
+        return tormaysResults
+    }
+
+    @PostMapping
+    fun createTormaysTarkasteluForHanke(@ValidHanke @RequestBody hanke: Hanke?): ResponseEntity<Any> {
+        logger.info {
+            "Creating tormaystarkastelu for hanke: ${hanke?.toLogString()}"
+        }
+
+        if (hanke == null) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(HankeError.HAI1002)
+        }
+
+        return try {
+
+            //if(hanke.tilat.onTiedotLiikenneHaittaIndeksille)
+            // then calculate:
+            // - call service to create tormaystarkastelu -> you will get TormaystarkasteluTulos
+            // - service has saved tormaystulos to database and saved hanke.onLiikenneHaittaIndeksi=true (or is it services job to do that?)
```

Comment:
One takes a Hanke-object from the HankeService (if not already have it from there, to ensure the flag states are current), manipulates the flag(s) as desired, then calls HankeService.updateHankeStateFlags(hanke).
See begin of HankeGeometriatServiceImpl.saveGeometriat() for an example how it is done.